_rules/aria-required-context-role-ff89c9.md

```diff
@@ -0,0 +1,229 @@
+
+```html
+<ul>
+	<div role="listitem">List item 1</div>
+</ul>
+```
+
+#### Passed Example 3
+
+Element contained within its [required context role](https://www.w3.org/TR/wai-aria-1.1/#scope) even though it is not a direct child of the [context role](https://www.w3.org/TR/wai-aria-1.1/#scope).
+
+```html
+<div role="list">
+	<div role="presentation">
+		<div role="listitem">List item 1</div>
+	</div>
+</div>
+```
+
+#### Passed Example 4
+
+`aria-owns` used to give the target element the right [context role](https://www.w3.org/TR/wai-aria-1.1/#scope).
+
+```html
+<div role="list" aria-owns="id1"></div>
+<div id="id1" role="listitem">List item 1</div>
+```
+
+#### Passed Example 5
+
+`aria-owns` trumps ownership by closest ancestor, giving the element with [explicit semantic role](#explicit-role) of `listitem` the correct [context role](https://www.w3.org/TR/wai-aria-1.1/#scope).
+
+```html
+<div role="list" aria-owns="id1">
+	<div role="navigation">
+		<div id="id1" role="listitem">List item 1</div>
+	</div>
+</div>
+```
+
+#### Passed Example 6
+
+Since implicit ownership can cross shadow boundaries, the element with the [explicit semantic role](#explicit-role) of `listitem` is contained within its [required context role](https://www.w3.org/TR/wai-aria-1.1/#scope) `list`.
+
+```html
+<div id="host" role="list"></div>
+
+<script>
+	const host = document.querySelector('#host')
+	const root = host.attachShadow({ mode: 'open' })
+	root.innerHTML = '<div role="listitem">List item 1</div>'
+</script>
+```
+
+### Failed
+
+#### Failed Example 1
+
+The `listitem` has no [context role](https://www.w3.org/TR/wai-aria-1.1/#scope).
+
+```html
+<div role="listitem">List item 1</div>
+```
+
+#### Failed Example 2
+
+The `listitem` is owned by the `tabpanel`, because it is the closest ancestor, but `tabpanel` is not the correct [context](https://www.w3.org/TR/wai-aria-1.1/#scope) for `listitem`.
+
+```html
+<div role="list">
+	<div role="tabpanel">
+		<div role="listitem">List item 1</div>
+	</div>
+</div>
+```
+
+#### Failed Example 3
+
+The `listitem` is owned by the `aria-label="menu"` div, rather than the `list`.
+
+```html
+<div role="list">
+	<div aria-label="menu">
+		<div role="listitem">List item 1</div>
+	</div>
+</div>
+```
+
+#### Failed Example 4
+
+Since explicit ownership cannot cross shadow boundaries, the element with the [explicit semantic role](#explicit-role) of `listitem` does not have a [context role](https://www.w3.org/TR/wai-aria-1.1/#scope).
+
+```html
+<div role="list" aria-owns="item"></div>
+
+<div id="host"></div>
+
+<script>
+	const host = document.querySelector('#host')
+	const root = host.attachShadow({ mode: 'open' })
+	root.innerHTML = '<div id="item" role="listitem">List item 1</div>'
+</script>
+```
+
+### Inapplicable
+
+#### Inapplicable Example 1
+
+There is no element with an [explicit semantic role](#explicit-role).
+
+```html
+<ul>
+	<li>List item 1</li>
+</ul>
+```
```

Comment:
I do not fully understand why we don't care about elements without explicit role.
Notably, I guess that
```html
<div role="tabpanel">
  <li>Foo</li>
</div>
```
is a pretty bad idea. This is currently inapplicable. Why don't we want to have that applicable (and failing)? What other problem would it cause elsewhere?

Reply:
This will be caught by the ARIA required owned element rule #245. The second part of that is, in HTML, the rules have to be a little different. In HTML, this isn't OK:

```
<ul>  <div role="none">  <li>...</li>  </div>  </ul>
```

That requires logic that I think is better put into a separate rule. Definitely not saying that shouldn't be tested, just that adding that kind of logic to this rule would make this more complicated than having separate rules.

Reply:
Hmm… As far as I understand #245 
```html
<div role="tablist">
  <div role="tab">Foo</div>
  <li>Bar</li>
</div>
```
will pass "ARIA required owned element" because the `tablist` has a `tab` child. And the `tab` will similarly pass this rule. And the `li` will be ignored by this rule (because it does not have an explicit role) and thus won't be caught by any rule currently.

Reply:
We'll look at this in the aria required owned element rule. That's not part of this rule.